src/main/java/org/generationcp/middleware/dao/oms/CVTermDao.java

```diff
@@ -1381,4 +1418,331 @@ public CVTerm save(final String name, final String definition, final CvId cvId)
 		return this.save(cvTerm);
 	}
 
+	/**
+	 * Count all variables in the database filtered by variableTypes.
+	 *
+	 * @param variableTypes
+	 * @return
+	 */
+	public long countAllVariables(final List<Integer> variableTypes) {
+		try {
+
+			final SQLQuery sqlQuery = this.getSession().createSQLQuery(this.createCountVariableQuery(false));
+			final List<String> variableTypeNames =
+				variableTypes.stream().map(i -> VariableType.getById(i).getName()).collect(Collectors.toList());
+			sqlQuery.setParameterList(VARIABLE_TYPE_NAMES, variableTypeNames);
+			return ((BigInteger) sqlQuery.uniqueResult()).longValue();
+
+		} catch (final HibernateException e) {
+			LOG.error(e.getMessage(), e);
+			throw new MiddlewareQueryException("Error in countAllVariables() query in CVTermDao: " + e.getMessage(), e);
+		}
+	}
+
+	/**
+	 * Gets the list of all variables in the database filtered by variableTypes.
+	 *
+	 * @param pageSize
+	 * @param pageNumber
+	 * @param variableTypes
+	 * @param cropName
+	 * @return
+	 */
+	public List<VariableDTO> getAllVariables(final Integer pageSize, final Integer pageNumber, final List<Integer> variableTypes,
```

Comment:
For uniformity and easier to understand code, place query parameters first before paging parameters.